### DISCUSSION TITLE
GitHub actions does not print git diff

### DISCUSSION THREAD

**Discussion by alonyb** [community/community#25186] [Actions]
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/1/120e4440251c7f2013920ede6d36964e9751423c.png" data-download-href="/uploads/short-url/2zJ9X3LGMk5gzdvo2cje0u3SKrW.png?dl=1" title="Screen Shot 2020-01-18 at 21.04.50.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/13793894/181089297-7c055733-ce36-4a81-9c64-d7534ae3a219.png" alt="Screen Shot 2020-01-18 at 21.04.50.png" data-base62-sha1="2zJ9X3LGMk5gzdvo2cje0u3SKrW" width="690" height="473" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/1/120e4440251c7f2013920ede6d36964e9751423c_2_690x473.png, https://aws1.discourse-cdn.com/github/optimized/2X/1/120e4440251c7f2013920ede6d36964e9751423c_2_1035x709.png 1.5x, https://aws1.discourse-cdn.com/github/original/2X/1/120e4440251c7f2013920ede6d36964e9751423c.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/1/120e4440251c7f2013920ede6d36964e9751423c_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Screen Shot 2020-01-18 at 21.04.50.png</span><span class="informations">1252×860 81.9 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>Why workflow does not print git diff command.</p>
<p>It’s a branch, but it should show branch differences.</p>

**Comment by BrightRan** [comment, marked as answer]
<p>Which version of the checkout action are you using in your workflow, <a href="https://github.com/actions/checkout/tree/v1" rel="nofollow noopener"><strong>v1</strong></a> or <a href="https://github.com/actions/checkout/tree/v2" rel="nofollow noopener"><strong>v2</strong></a>?</p>
<p>There are few notes needd you pay attention to:</p>
<p>1. By default <strong>git diff</strong> will show you any uncommitted changes since the last commit.</p>
<ol start="2">
<li>When using <strong>actions/checkout@v2</strong> , it fetches only a single commit by default.</li>
</ol>
<p>If you want to view the differences between the currently latest commit and the last commit, you can try to use  <strong>actions/checkout@v1</strong>  to checkout the repository, and use teh following command to view the differences.</p>
<pre><code>git diff --name-only --diff-filter=AM ${{ github.event.before }} ${{ github.sha }}
</code></pre>

**Comment by alonyb** [comment]
<p>Hey, <a class="mention" href="/brightran">@brightran</a> I didn’t think in checkout version. But you are right, thank you a lot. :slight_smile:<br>
I will read the differences between them</p>

**Comment by DuncanRobotEngineer** [comment]
<p>Hi I am having some issues with a reference error on my node js script file that I am trying to resolve, can you guys assist on how I can resolve the error. I am new on node js and javascript thanks. I know this may not related to what I am replying to. </p>
<p>Here is the code below:</p>
<p>‘use strict’;<br>
const request = require(‘request’);<br>
const queryParams = { searchPhrase: ‘Andrew’ };<br>
const url = ‘<a href="https://api.com/v1/people" rel="nofollow noopener">https://api.com/v1/people</a>’;<br>
describe(‘api test’, () =&gt; {<br>
it(‘test case 1’, () =&gt; {<br>
const response = request.get(url, queryParams);<br>
const personId = response.results[2].id;<br>
const deleteResponse = request.delete(<code>url/${personId}</code>);<br>
expect(deleteResponse.statusCode).to.eql(204);<br>
const afterDeleteGet = request.get(<code>url/${personId}</code>);<br>
expect(afterDeleteGet.statusCode).to.eql(404);<br>
});<br>
});</p>

**Comment by hannahgreene** [comment]
<p>How do you diff with master? Instead of comparing </p>
<pre><code>${{ github.sha}} ${{ github.event.before}}
</code></pre>
<p>I want to compare </p>
<pre><code>${{ github.sha}}
</code></pre>
<p>to master</p>
